src/main/java/chessgame/domain/piece/Pieces.java

```diff
@@ -79,18 +77,22 @@ public boolean isCheckmate(final Color color) {
         if (!canRemovable(kingPoint, color)) {
             return false;
         }
-        return findPieceWithPoint(kingPoint)
-                .orElseThrow(IllegalStateException::new)
-                .findLegalMovePoints(this)
+        return getLegalMovePoints(kingPoint)
                 .stream()
                 .allMatch(point -> canRemovable(point, color));
     }
 
+    private Set<Point> getLegalMovePoints(Point point) {
+        return findPieceWithPoint(point)
+                .orElseThrow(IllegalStateException::new)
+                .findLegalMovePoints(this);
```

Comment:
```suggestion
       final Piece piece = findPieceWithPoint(point)
            .orElseThrow(IllegalStateException::new);
       return piece.findLegalMovePoints(this);
```
저라면 이렇게 작성했을 것 같아요! orElseThrow를 통해 종단 연산이 수행됐는데, 또다른 메서드 체이닝이 있어서 읽는 사람이 조금 헷갈릴 것 같거든요 😭

Reply:
아 코멘트의 의도를 완전 잘못 이해했던것 같습니다. 전혀 생각하지 못했던 것인데. 썬처럼 생각하는 것이 합리적인것같습니다. 바로 적용하겠습니다!